File: programmers/etc/level2/숫자_블록.py
def maxdivisor(n):
    if n == 1:
        return 0
    divisors = []
    for x in range(1, int(n**0.5)+1):
        if n%x == 0:
            if x <= 1e7:
                divisors.append(x)
            if n//x <= 1e7 and n//x != n:
                divisors.append(n//x)
    return max(divisors)

def solution(begin, end):
    return [maxdivisor(n) for n in range(begin, end+1)]


begin, end = 1, 10
print(solution(begin, end)) # [0, 1, 1, 2, 1, 3, 1, 4, 3, 5]

File: Java/programmers/etc/level2/숫자블록.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 숫자블록 {
    public int[] solution(long begin, long end) {
        int first = (int) begin;
        int last = (int) end;
        int sz = last-first+1;
        int[] answer = new int[sz];
        
        for (int i=first; i<=last; i++) {
            answer[i-first] = maxdivisor(i);
        }

        return answer;
    }

    private static int maxdivisor(int n) {
        if (n == 1) return 0;
        List<Integer> divisors = new ArrayList<>();
        for (int x=1; x<(int)Math.sqrt(n)+1; x++) {
            if (n%x == 0) {
                if (x <= 10_000_000) {
                    divisors.add(x);
                }
                if (n/x <= 10_000_000 && n/x != n) {
                    divisors.add(n/x);
                }
            }
        }
        return Collections.max(divisors);
    }
}
